exec-service-wasmer/src/wasmer_executor.rs

```diff
@@ -1,101 +1,62 @@
+use crate::wasmer_logger as WasmerLogger;
 use crate::WasmerInstance;
+use log::info;
 use multiversx_vm_executor::{
     CompilationOptions, Executor, ExecutorError, Instance, OpcodeCost, ServiceError, VMHooks,
 };
+use std::cell::RefCell;
 use std::ffi::c_void;
 use std::rc::Rc;
-use std::sync::Arc;
-
-pub struct WasmerExecutor {
-    pub data: Rc<WasmerExecutorData>,
-}
-
-#[derive(PartialEq, PartialOrd, Eq)]
-pub enum WasmerExecutorLogLevel {
-    None,
-    Debug,
-    Trace,
-}
-
-impl TryFrom<u64> for WasmerExecutorLogLevel {
-    type Error = &'static str;
-
-    fn try_from(value: u64) -> Result<Self, Self::Error> {
-        match value {
-            0 => Ok(WasmerExecutorLogLevel::None),
-            1 => Ok(WasmerExecutorLogLevel::Debug),
-            2 => Ok(WasmerExecutorLogLevel::Trace),
-            _ => Err("WasmerExecutor undefined log level"),
-        }
-    }
-}
+use std::sync::{Arc, Mutex};
 
 pub struct WasmerExecutorData {
     pub vm_hooks: Rc<Box<dyn VMHooks>>,
-    pub opcode_cost: Arc<OpcodeCost>,
-    pub log_level: WasmerExecutorLogLevel,
+    pub opcode_cost: Arc<Mutex<OpcodeCost>>,
 }
 
 impl WasmerExecutorData {
-    pub(crate) fn debug(&self, message: &str) {
-        if self.log_level >= WasmerExecutorLogLevel::Debug {
-            println!("[DEBUG] {message}");
+    fn set_vm_hooks_ptr(&mut self, vm_hooks_ptr: *mut c_void) -> Result<(), ExecutorError> {
+        if let Some(vm_hooks) = Rc::get_mut(&mut self.vm_hooks) {
+            vm_hooks.set_vm_hooks_ptr(vm_hooks_ptr);
+            return Ok(());
         }
```

Comment:
you can use else and not need to `return`

Reply:
Done